consult.el

```diff
@@ -2149,5 +2149,87 @@ if not available use `default-directory'."
 
 (add-hook 'consult--completion-refresh-hook #'consult--icomplete-refresh)
 
+
+;;;; find, fd, locate
+(defun consult--guess-initial-dir (&optional askdir)
```

Comment:
Instead of this function you should `(funcall consult-grep-directory-function)`. We should rename `consult-grep-directory-function` to something more generic though.

Reply:
Uhmmm... I hadn't noticed this function before.
Yes, I think we should unify `consult-grep-directory-default` and `consult--guess-initial-dir` under a generic name. Please choose one.
Doesn't my implementation be enough?

Reply:
I had something only for grep before, but I made it more generic in https://github.com/minad/consult/commit/fef1fc8522281d17d79c4605e83e4c8ee6aea471.